vimeo-networking/src/main/java/com/vimeo/networking/utils/VimeoNetworkUtil.java

```diff
@@ -209,30 +172,62 @@ public void run() {
     }
 
     /**
-     * This utility method takes a Retrofit response and extracts a {@link VimeoError} object out of it if
+     * Creates a local error with the provided error message.
+     *
+     * @param message The message explaining the error.
+     */
+    @NotNull
+    public static VimeoResponse.Error.Exception createLocalError(@NotNull final String message) {
+        return new VimeoResponse.Error.Exception(new IllegalArgumentException(message));
+    }
+
+    /**
+     * Creates a local API error with the provided error message and invalid parameters list.
+     *
+     * @param message           The message explaining the error.
+     * @param invalidParameters The parameters that were invalid.
+     */
+    @NotNull
+    public static VimeoResponse.Error.Api createLocalApiError(@NotNull final String message,
+                                                              @NotNull final InvalidParameter... invalidParameters) {
+        return new VimeoResponse.Error.Api(ApiErrorFactory.createApiErrorWithInvalidParameters(message,
+                                                                                               invalidParameters),
+                                           -1);
+    }
+
+    /**
+     * This utility method takes a Retrofit response and extracts a {@link ApiError} object out of it if
      * applicable. It will return null in the case where there has been a successful response.
      *
      * @param response A non-null response from the Vimeo API
-     * @return a {@link VimeoError} object extracted from the response or null
+     * @return a {@link VimeoResponse.Error} object extracted from the response or null
      */
-    @Nullable
-    public static <ResponseType_T> VimeoError getErrorFromResponse(@Nullable final Response<ResponseType_T> response) {
-        if (response != null && response.isSuccessful()) {
-            return null;
+    @NotNull
+    public static <ResponseType_T> VimeoResponse.Error getErrorFromResponse(@NotNull final Response<ResponseType_T> response) {
+        if (response.isSuccessful()) {
+            throw new IllegalStateException("Cannot get error from a successful response");
```

Comment:
Like this a lot more than returning null.

Reply:
Yeah, I looked at all the usage of it and in every case we were checking whether or not the response was unsuccessful before calling the function, so failing like this makes more sense.